client/service/src/builder.rs

```diff
@@ -903,8 +949,33 @@ where
 		Arc::new(TransactionPoolAdapter { pool: transaction_pool, client: client.clone() }),
 		config.prometheus_config.as_ref().map(|config| &config.registry),
 	)?;
-
 	spawn_handle.spawn("network-transactions-handler", Some("networking"), tx_handler.run());
+
+	// crate statement gossip protocol and add it to the list of supported protocols of
+	// `network_params`
+	if let Some(statement_store) = statement_store {
+		let statement_protocol_executor = {
+			let spawn_handle = Clone::clone(&spawn_handle);
+			Box::new(move |fut| {
+				spawn_handle.spawn("network-statement-validator", Some("networking"), fut);
+			})
+		};
+		let statement_handler_proto = statement_handler_proto
+			.expect("statement_handler_proto is always created when statement_store is `Some`");
+		let statement_handler = statement_handler_proto.build(
+			network.clone(),
+			sync_service.clone(),
+			statement_store.clone(),
+			config.prometheus_config.as_ref().map(|config| &config.registry),
+			statement_protocol_executor,
+		)?;
+		spawn_handle.spawn(
+			"network-statement-handler",
+			Some("networking"),
+			statement_handler.run(),
+		);
+	}
+
```

Comment:
Should as well not be done here. This entire logic could just be part of the statement crate?

Reply:
I took the same design choices as transaction pool. The service is there to tie the client components and networking, isn't it?

Reply:
We created this huge monster, which was a "failure". If someone wants this, they can just import it in their own service.

Reply:
Moved all that stuff to node service